src/adera/Machines/Rocket.h

```diff
@@ -103,21 +117,25 @@ class MachineRocket : public osp::active::Machine
     osp::active::WireInput m_wiThrottle { this, "Throttle" };
 
     osp::active::ActiveEnt m_rigidBody  { entt::null };
-    float m_thrust{0.0f};
+    fuel_list_t m_resourceLines;
+
+    Parameters m_params;
 }; // MachineRocket
 
-inline MachineRocket::MachineRocket(float thrust)
- : Machine(true)
- , m_thrust(thrust)
+inline MachineRocket::MachineRocket(Parameters params, fuel_list_t& resources)
```

Comment:
don't take by l-value reference if you are going to unconditionally std::move. Just take the parameter by value in that case.